### PR TITLE
Meta: Ensure features are described only once in the readme

### PR DESCRIPTION
**Pull request by gamemaker1**
## Related Issues

Mentioned in https://github.com/refined-github/refined-github/pull/4699#issuecomment-962593291.

## What this PR does

It ensures that all features are described only once in [`readme.md`](readme.md).

### CODE REVIEW

File: build/verify-features.ts
Comment (gamemaker1) [inline build/verify-features.ts:71]: I'll try adding support for parsing the 'highlight'ed features as well

Comment (gamemaker1) [reply build/verify-features.ts]: done!

File: build/verify-features.ts
Comment (yakov116) [inline build/verify-features.ts:2]:
```suggestion
import regexJoin from 'regex-join';
import {existsSync, readdirSync, readFileSync} from 'node:fs';
```

Lint :)

**Comment by yakov116** [comment]
@gamemaker1 are you able to give the line number in the console? (Only if very simple no need to work hard on it)

**Comment by gamemaker1** [comment]
Will try it out

**Comment by gamemaker1** [comment]
> Line numbers in the console

Done!

File: build/verify-features.ts
Comment (fregante) [inline build/verify-features.ts:66]: We already have the regexes somewhere, we should reuse them.

File: build/verify-features.ts
Comment (fregante) [inline build/verify-features.ts:77]: I don’t think we need the like number, that's easy to find once you know there's a duplicate

Comment (gamemaker1) [reply build/verify-features.ts]: Commented it out for now, let me know if I should remove it completely or keep it.

IMO it's a good feature, saves the effort of a manual ctrl/cmd+f.

File: build/verify-features.ts
Comment (gamemaker1) [reply build/verify-features.ts]: Done

**Comment by gamemaker1** [comment]
Seems good to go... any other changes to be made?

File: build/verify-features.ts
Comment (cheap-glitch) [inline build/verify-features.ts:72]: I think unused code should be removed. If we change our minds, we can always get it back from this PR's commit.

Comment (gamemaker1) [reply build/verify-features.ts:72]: I just commented it out for now as I was waiting on the decision to include line numbers. If we decide to keep it, I will uncomment that part, else I will delete it completely.

File: build/verify-features.ts
Comment (fregante) [inline build/verify-features.ts:66]: Why?

File: build/verify-features.ts
Comment (fregante) [inline build/verify-features.ts:71]:
```suggestion
	if (featureMatch?.length + highlightedFeatureMatch?.length > 1) {
```

If TS complains, add a couple of `Number()` calls

**Review by fregante — APPROVED** [review]
It can be merged after the 3 open reviews are fixed

File: build/verify-features.ts
Comment (gamemaker1) [reply build/verify-features.ts:66]: `match` is more concise in this case, as we just need to call it once to get all the matches. `exec` will need us to use a while loop to get all the matches.

File: build/verify-features.ts
Comment (gamemaker1) [reply build/verify-features.ts:71]: Used `(featureMatch?.length ?? 0) + (highlightedFeatureMatch?.length ?? 0) > 1`

File: build/verify-features.ts
Comment (gamemaker1) [reply build/verify-features.ts]: `exec` seems to be more efficient when I'm counting line numbers too... so I'm using `exec` now.

File: build/verify-features.ts
Comment (gamemaker1) [reply build/verify-features.ts]: I've kept the line number functionality, see my comment below.

**Comment by gamemaker1** [comment]
I rewrote it to use `regexp.exec` instead of `string.match` - the linter was correct after all ;) - and it was much cleaner than using `string.match`. Line numbers for the position of descriptions are also printed in case of duplicates. The following cases are checked:

- If a feature is described twice in the feature list
- If a feature is part of the highlighted features and part of the feature list

The following case, however, is not checked:

- If a feature is described twice within the highlights section

I am trying to add that right now.

File: build/verify-features.ts
Comment (gamemaker1) [inline build/verify-features.ts:81]: For some reason this regex returns matches only if the feature has been described more than once in the highlights section... not sure why that is happening

Comment (gamemaker1) [reply build/verify-features.ts:81]: The other regex (for the list of features) is matching all the descriptions

**Review by fregante — CHANGES_REQUESTED** [review]
> I've kept the line number functionality, see my comment below.

I had already reviewed and rejected this part

**Comment by gamemaker1** [comment]
Removed the line number functionality, please let me know if there are any other changes I should make.

**Comment by fregante** [comment]
Thank you!

**Pull request merged**
